File: leetcode/1252.奇数值单元格的数目.cpp
#include <iostream>
#include <vector>
using namespace std;
/*
 * @lc app=leetcode.cn id=1252 lang=cpp
 *
 * [1252] 奇数值单元格的数目
 */

// @lc code=start
class Solution
{
public:
    int oddCells(int m, int n, vector<vector<int>> &indices)
    {
        int res = 0;
        int cache_m[m];
        memset(cache_m, 0, sizeof(cache_m));
        int cache_n[n];
        memset(cache_n, 0, sizeof(cache_n));
        for (auto &&i : indices)
        {
            cache_m[i[0]]++;
            cache_n[i[1]]++;
        }
        for (size_t r = 0; r < m; r++)
        {
            for (size_t c = 0; c < n; c++)
            {
                res += (cache_m[r] + cache_n[c]) % 2;
            }
        }
        return res;
    }
};
// @lc code=end
int main()
{
    Solution s;
    vector<vector<int>> vect;
    vector<int> vect1;
    vect1.push_back(0);
    vect1.push_back(1);
    vect.push_back(vect1);
    vector<int> vect2;
    vect2.push_back(1);
    vect2.push_back(1);
    vect.push_back(vect2);
    cout << s.oddCells(2, 3, vect) << endl;
}

File: leetcode/1252.奇数值单元格的数目.py
#
# @lc app=leetcode.cn id=1252 lang=python3
#
# [1252] 奇数值单元格的数目
#
from typing import List


# @lc code=start
class Solution:
    def oddCells(self, m: int, n: int, indices: List[List[int]]) -> int:
        cache_r = [0] * m
        cache_c = [0] * n
        for r, c in indices:
            cache_r[r] += 1
            cache_c[c] += 1
        return sum(
            (cache_r[r] + cache_c[c]) % 2
            for r in range(m)
            for c in range(n)
        )


# @lc code=end
print(Solution().oddCells(m=2, n=3, indices=[[0, 1], [1, 1]]))
print(Solution().oddCells(m=2, n=2, indices=[[1, 1], [0, 0]]))
